instana/span.py

```diff
@@ -247,24 +248,23 @@ def __init__(self, span, source, service_name, **kwargs):
         super(RegisteredSpan, self).__init__(span, source, service_name, **kwargs)
         self.n = span.operation_name
         self.k = 1
-        if span.operation_name in self.ENTRY_SPANS:
+
+        if span.operation_name in self.ENTRY_SPANS or (
+                span.operation_name == "rabbitmq" and span.tags.get('sort', None) == "consume"):
             # entry
             self._populate_entry_span_data(span)
             self.data["service"] = service_name
         elif span.operation_name in self.EXIT_SPANS:
-            self.k = 2 # exit
+            self.k = 2  # exit
             self._populate_exit_span_data(span)
         elif span.operation_name in self.LOCAL_SPANS:
-            self.k = 3 # intermediate span
+            self.k = 3  # intermediate span
             self._populate_local_span_data(span)
 
-        if "rabbitmq" in self.data and self.data["rabbitmq"]["sort"] == "consume":
```

Comment:
@pdimitra:

I checked `span.py` and it seems like the `_populate_entry_span_data` & `_populate_exit_span_data` for `rabbitmq` is the same. In this case, we can let it stay in the `ENTRY_SPANS` tuple and just change the method here to `publish` and set `self.k = 2`